src/index.ts

```diff
@@ -2,87 +2,121 @@ import cp = require('child_process')
 import fs = require('fs')
 import p = require('path')
 
-// Logger
-const l = (msg: string): void => console.log(`husky - ${msg}`)
-
-// Git command
-const git = (args: string[]): cp.SpawnSyncReturns<Buffer> =>
-  cp.spawnSync('git', args, { stdio: 'inherit' })
-
-export function install(dir = '.husky'): void {
-  // Ensure that we're inside a git repository
-  // If git command is not found, status is null and we should return.
-  // That's why status value needs to be checked explicitly.
-  if (git(['rev-parse']).status !== 0) {
-    return
-  }
-
-  // Custom dir help
-  const url = 'https://git.io/Jc3F9'
-
-  // Ensure that we're not trying to install outside of cwd
-  if (!p.resolve(process.cwd(), dir).startsWith(process.cwd())) {
-    throw new Error(`.. not allowed (see ${url})`)
-  }
-
-  // Ensure that cwd is git top level
-  if (!fs.existsSync('.git')) {
-    throw new Error(`.git can't be found (see ${url})`)
-  }
-
-  try {
-    // Create .husky/_
-    fs.mkdirSync(p.join(dir, '_'), { recursive: true })
-
-    // Create .husky/_/.gitignore
-    fs.writeFileSync(p.join(dir, '_/.gitignore'), '*')
+interface Options {
```

Comment:
In your 2nd commit you switched from the name Logger to Options

Having recently worked with pino, I think Logger is a better name (please mind that I'm just casually reviewing here)

Reply:
That was my intention at first, the initial draft of the PR was in fact providing a way to pass a custom logger.

Then I realized that if one wanted later to add other options apart from logging functions, the `configure` function would take multiple arguments: `configure(logger: Logger = defaultLogger, customOptions: CustomOptions = {})`.

I found this rather awkward to separate optional custom logging functions from other potential future options.

I know it's a rather opinionated choice, perhaps having multiple opinions from different people on this matter might help?

Reply:
Upon re reading this I agree "Options" is a better name